builder/frameworks/espidf.py

```
@@ -1461,9 +1461,55 @@ def prepare_build_envs(config, default_env, debug_allowed=True):
+    populate_idf_env_vars(idf_env)
+    NINJA_DIR = platform.get_package_dir("tool-ninja")
+    ninja_exe = os.path.join(NINJA_DIR, "ninja")
+    for ninja_target, src_path in generated_targets:
+        print("Generating source: %s" % src_path)
+        result = exec_command(
+            [ninja_exe, "-C", BUILD_DIR, ninja_target],
+            env=idf_env,
+        )
+        if result["returncode"] != 0:
+            sys.stderr.write(
+                "Error: Failed to generate source file '%s'\n" % target
```

Ruff (0.15.7)

[error] Undefined name `target`

(F821)